### PR TITLE
clean up scope unit tests

### PR DESCRIPTION
**Pull request by nojnhuh**
**What type of PR is this?**

**What this PR does / why we need it**:

This PR removes most usage of our constructor wrappers to create Scope objects in unit tests in order to avoid the extra setup that requires to create a valid reference to an existing AzureClusterIdentity. That adds quite a few moving parts when all we are trying to test is a function like this that doesn't actually use the identity at all:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/b8f23bd30a7325f163bf0baa8c2201a594e3a971/azure/scope/cluster.go#L636-L645

I'm planning to make some changes to the identity stuff soon for #5017 and #1077 that would require some change to all of these tests, but with this change it shouldn't need any extra changes. Hopefully this prevents us from needing other sweeping changes to these tests for unrelated changes like this:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4208/files#diff-bae329736fc788cebbe920f1680bfd39eb52e6f1bdaaa0a620c60037d3840462

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [X] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: azure/scope/managedcontrolplane_test.go [hunk 1]
```diff
@@ -272,28 +265,10 @@ func TestManagedControlPlaneScope_PoolVersion(t *testing.T) {
 	}
 
 	for _, c := range cases {
-		c := c
```
Comment (mboersma) [inline azure/scope/managedcontrolplane_test.go:275]: I guess we can ditch this now with Go 1.22?

Comment (nojnhuh) [reply azure/scope/managedcontrolplane_test.go:275]: I think this change was probably safe even before that since these tests aren't running in parallel.

**Review by willie-yao — APPROVED** [review]
/approve

**Pull request merged**